include/net/nrf_cloud.h

```diff
@@ -673,6 +673,28 @@ struct nrf_cloud_init_param {
 	const char *application_version;
 };
 
+enum nrf_cloud_fota_reboot_status {
+	FOTA_REBOOT_REQUIRED,	/** Reboot to install update. */
+	FOTA_REBOOT_SUCCESS,	/** Reboot to complete FOTA job. */
+	FOTA_REBOOT_FAIL,	/** Reboot because FOTA job failed. */
+	FOTA_REBOOT_SYS_ERROR	/** Reboot because fatal system error occurred. */
+};
+
+/**
+ * @brief  Event handler registered with the module to handle asynchronous
+ * events from the module.
+ *
+ * @param[in]  evt The event and any associated parameters.
+ */
+typedef void (*fota_reboot_handler_t)(enum nrf_cloud_fota_reboot_status status);
+
+struct nrf_cloud_fota_support_ctx {
+	struct nrf_cloud_rest_context *rest_ctx;
+	const char *device_id;
```

Comment:
could the device_id buffer live in fota_support? 
and then `nrf_cloud_client_id_get()` could be called in `nrf_cloud_fota_support_init()`?

Reply:
Yes, I could do that.  I did it this way to eliminate duplication of effort, but if you think moving it here is more important let me know.

Reply:
if an application is only using nrf cloud for REST FOTA, i don't think it would be using the device id for anything else.

if an application is using other nrf cloud REST services, the fota support module, in its current form, might be a little awkward to use since the process function is blocking and using the shared rest context.

Reply:
Your `nrf_cloud_rest_fota` sample prints out the device id; that's why I decided to reuse it rather than fetch it again.

The `nrf_cloud_fota_support_process()` function returns at the same points your code in `main()` used to loop around from.  If there is no update, for example, it returns right away.

It sounds like you would have preferred that the 8 smaller functions that `nrf_cloud_fota_support_process()` calls be made public.  That seems like a bigger maintenance burden than packaging it as I have.  Besides, we don't recommend people use REST for multiple services.

Reply:
`Your nrf_cloud_rest_fota sample prints out the device id; that's why I decided to reuse it rather than fetch it again.`
ok, leave it up to the application.

`It sounds like you would have preferred that the 8 smaller functions `
fota_support in its current form is fine for the use case of doing only REST based FOTA in the sample, so for this ticket, no i would not prefer it be separated.

`we don't recommend people use REST for multiple services.`
interesting... i think there are use-cases which would allow it, but no need to discuss that here.